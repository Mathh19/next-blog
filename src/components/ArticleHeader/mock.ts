import { ArticleHeaderProps } from '.';

export default {
  title:
    'Sint deserunt officia sit ipsum. Minim commodo sit ea labore sit sit ex ad tempor esse labore.',
  excerpt:
    'Proident sunt est enim fugiat sit incididunt sunt veniam adipisicing amet aliquip.',
  cover:
    'https://res.cloudinary.com/dlizakp2a/image/upload/v1614696630/Frozen_sunset_on_the_lake_by_Manuel_Arslanyan_9f9cd8ea10.jpg',
  createdAt: '2022-09-02T15:16:09.246+00:00',
  author: {
    id: '123456789',
    attributes: {
      slug: 'matheus-freitas',
      name: 'Matheus Freitas',
    },
  },
  categories: [
    {
      id: 'asdslgyh65',
      attributes: {
        name: 'React',
        slug: 'react',
      },
    },
    {
      id: 'asdslgyh66',
      attributes: {
        name: 'JavaScript',
        slug: 'javascript',
      },
    },
  ],
} as ArticleHeaderProps;
