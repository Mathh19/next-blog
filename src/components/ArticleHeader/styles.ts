import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacings.xlarge};
    margin-bottom: ${theme.spacings.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    ${HeadingStyles} {
      margin: 0;
      margin-bottom: ${theme.spacings.medium};
    }
  `}
`;

export const Excerpt = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.medium} 0;
    font-size: ${theme.font.sizes.medium};
  `}
`;

export const CoverContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: center;
    margin-bottom: ${theme.spacings.medium};
  `}
`;

export const Cover = styled.img`
  ${css`
    max-width: 100%;
    display: block;
  `}
`;
