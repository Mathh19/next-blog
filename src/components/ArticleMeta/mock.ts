import { ArticleMetaProps } from '.';

export default {
  createdAt: '2022-09-02T15:16:09.246+00:00',
  author: {
    id: '1234567890',
    attributes: {
      slug: 'matheus-freitas',
      name: 'Matheus Freitas',
    },
  },
  categories: [
    {
      id: 'asdslgyh65',
      attributes: {
        name: 'React',
        slug: 'react',
      },
    },
    {
      id: 'asdslgyh66',
      attributes: {
        name: 'JavaScript',
        slug: 'javascript',
      },
    },
  ],
} as ArticleMetaProps;
