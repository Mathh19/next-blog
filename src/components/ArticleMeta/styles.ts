import styled, { css } from 'styled-components';

export const Wrapper = styled.article`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.darkGray};
    font-style: italic;

    .categories span::after {
      content: ', ';
    }
    .categories span:last-child::after {
      content: '';
    }

    a {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 300ms ease-in-out;

      &:hover {
        filter: brightness(50%);
      }
    }
  `}
`;
