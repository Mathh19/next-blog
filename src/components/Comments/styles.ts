import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    max-width: ${theme.sizes.content};
    width: 100%;
    margin: ${theme.spacings.large} auto;
    padding: 0 ${theme.spacings.large};
  `}
`;

export const NoComments = styled.div`
  ${({ theme }) => css`
    text-align: center;
    color: ${theme.colors.secondary};
    font-size: ${theme.font.sizes.medium};
  `}
`;
