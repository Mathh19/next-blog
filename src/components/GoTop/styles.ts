import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    background-color: #082032;
    color: ${theme.colors.secondary};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    bottom: 2rem;
    right: 2rem;
    z-index: 6;
    opacity: 0.7;
    border-radius: 50%;
    border: 0.2rem solid ${theme.colors.secondary};
    &:hover {
      transition: 500ms ease-in-out all;
      background-color: #12324a;
      opacity: unset;
    }
  `}
`;
