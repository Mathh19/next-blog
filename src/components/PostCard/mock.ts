export default {
  id: '1',
  attributes: {
    slug: 'blog-sobre-nextjs',
    title: 'Blog sobre Nextjs',
    excerpt: 'Esse Ã© um blog que fala um pouco sobre Next e como ele funciona.',
    cover: {
      id: '3',
      attributes: {
        alternativeText: 'next-capa.png',
        url: 'https://res.cloudinary.com/ddiiakz1t/image/upload/v1662140975/next_capa_ba07b5ec5d.png',
      },
    },
  },
};
