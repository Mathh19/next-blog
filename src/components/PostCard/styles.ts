import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    ${HeadingStyles} {
      margin: 0;
      margin-top: 1rem;
      margin-bottom: ${theme.spacings.small};
    }

    a {
    text-decoration: none;
    color: inherit;
    transition: all 300ms ease-in-out;
  }

  &:hover a {
    color: ${theme.colors.secondary};
  }

  &:hover img {
    opacity: 0.8;
  }
  `}
`;

export const Cover = styled.img`
    ${({ theme }) => css`
    max-width: 100%;
    transition: opacity 300ms ease-in-out;
    border: 1px solid ${theme.colors.primary};
    `}
`;

export const Excerpt = styled.p``;
