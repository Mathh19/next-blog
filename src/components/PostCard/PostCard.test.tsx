import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { PostCard, PostCardProps } from '.';

import mock from './mock';

const props: PostCardProps = mock;

describe('<PostCard />', () => {
  it('should render a heading, cover and excerpt', () => {
    renderTheme(<PostCard {...props} />);

    expect(
      screen.getByRole('heading', { name: mock.data.attributes.title }),
    ).toBeInTheDocument();
    expect(
      screen.getByRole('img', { name: mock.data.attributes.title }),
    ).toBeInTheDocument();
    expect(screen.getByText(mock.data.attributes.excerpt)).toBeInTheDocument();
    expect(
      screen.getAllByRole('link', { name: mock.data.attributes.title })[0],
    ).toHaveAttribute('href', `/post/${mock.data.attributes.slug}`);
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<PostCard {...props} />);

    expect(container).toMatchSnapshot();
  });
});
