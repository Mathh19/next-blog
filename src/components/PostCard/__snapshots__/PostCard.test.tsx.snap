// Vitest Snapshot v1

exports[`<PostCard /> > should match snapshot 1`] = `
.c3 {
  color: #082032;
  line-height: 1.2;
  font-size: 2.4rem;
  text-transform: none;
}

.c0 .c2 {
  margin: 0;
  margin-top: 1rem;
  margin-bottom: 1.6rem;
}

.c0 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0:hover a {
  color: #DC143C;
}

.c0:hover img {
  opacity: 0.8;
}

.c1 {
  max-width: 100%;
  -webkit-transition: opacity 300ms ease-in-out;
  transition: opacity 300ms ease-in-out;
}

<div>
  <div
    class="c0"
  >
    <a
      href="/post/blog-sobre-nextjs"
    >
      <img
        alt="Blog sobre Nextjs"
        class="c1"
        src="https://res.cloudinary.com/ddiiakz1t/image/upload/v1662140975/next_capa_ba07b5ec5d.png"
      />
    </a>
    <h2
      class="c2 c3"
    >
      <a
        href="/post/blog-sobre-nextjs"
      >
        Blog sobre Nextjs
      </a>
    </h2>
    <p
      class=""
    >
      Esse Ã© um blog que fala um pouco sobre Next e como ele funciona.
    </p>
  </div>
</div>
`;
